task2/index.js

```diff
@@ -0,0 +1,25 @@
+function storeDetails(){  
+    var name=document.getElementById("username").value;  
+    var password=document.getElementById("password").value;  
+    var lowerCaseLetters = /[a-z]/g;
+    var upperCaseLetters = /[A-Z]/g;
+    var numbers = /[0-9]/g;
+    if(name.length == 0 || password.length == 0 || password.length<8){
```

Comment:
A more generic and better way to add validation can be to create a custom function and then call it. Something like this:

```
const validateInputs = (name, password) => {
    let inputIsValidated = true; // lets consider that inputs are fine and correct initially.

    if(!!name.length && (!!password.length || password.length < 8)){
        // Assuming that both name and password are required, there will be 2 conditions to check individually:
        // 1. name should not be empty
        // 2. password should not be empty OR password length should not be less than 8 characters
        // If any of the required inputs are not provided, it will result in validation being false.
        inputIsValidated = false;
    };

    return inputIsValidated;
}

const validationPassed = validateInputs(name, password);

if(!validationPassed) { 
    window.location.href="./index.html";
    window.location.reload(true)
    window.location.replace("./index.html")
    alert("please fill in email & password")
    .......
}
```

You can also modify the above method to return custom error messages as well.

@shubhamshukla91

Reply:
okay,